samples/gatsby-preview/src/index.jsx

```diff
@@ -0,0 +1,144 @@
+class App extends React.Component {
+  constructor(props) {
+    super(props)
+
+    const parameters = this.props.sdk.parameters
+    const projectId = parameters.installation.projectId
+
+    const { contentTypeSlug, isAutoUpdate } = parameters.instance
+
+    this.state = {
+      projectId,
+      projectUrl:
+        "https://" + projectId + "-preview.gtsb.io/" + contentTypeSlug + "/",
+      isAutoUpdate: isAutoUpdate,
+      webhookUrl:
+        "https://backends.ctffns.net/gatsby-preview-proxy/" + projectId
+    }
+  }
+
+  componentDidMount = () => {
+    this.detachFn = this.props.sdk.entry.onSysChanged(this.onSysChanged)
+
+    this.props.sdk.window.startAutoResizer()
+  }
+
+  componentWillUnmount = () => {
+    this.detachFn()
+    if (this.debounceInterval) {
+      clearInterval(this.debounceInterval)
+    }
+  }
+
+  onError = error => {
+    this.setState({ working: false })
+    this.props.sdk.notifier.error(error.message)
+  }
+
+  onSysChanged = () => {
+    if (!this.state.isAutoUpdate) {
+      return
+    }
+    if (this.debounceInterval) {
+      clearInterval(this.debounceInterval)
+    }
+    this.debounceInterval = setInterval(this.refreshGatsbyPreview, 1000)
+  }
+
+  refreshGatsbyPreview = () => {
+    if (this.debounceInterval) {
+      clearInterval(this.debounceInterval)
+    }
+
+    fetch(this.state.webhookUrl, {
+      method: "POST",
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify({})
+    }).then(
+      success => this.props.sdk.notifier.success("Gatsby Preview Updated!"),
+      error => this.props.sdk.notifier.error(error)
```

Comment:
I'd just hardcode some string ("failed to update"). Error will be an object so you'll get an notification with `[object Object]`